chia/rpc/wallet_rpc_api.py

```diff
@@ -1363,7 +1365,14 @@ async def nft_mint_nft(self, request) -> Dict:
             ]
         )
         fee = uint64(request.get("fee", 0))
-        spend_bundle = await nft_wallet.generate_new_nft(metadata, royalty_puzhash, target_puzhash, fee=fee)
+        spend_bundle = await nft_wallet.generate_new_nft(
+            metadata,
+            royalty_puzhash,
+            target_puzhash,
+            uint16(request.get("royalty_percentage", 0)),
+            request.get("use_did", True),
+            fee,
```

Comment:
If the minting wallet doesn't have a DID associated with it, should it be possible to set the royalty info?

Also, should `request.get("use_did", True),` be something like `request.get("use_did", wallet.get_did() != None)` to handle the same case where the NFT wallet doesn't have a DID?

Reply:
It doesn't matter here. If the DID is none, the use_did will be ignored.